lib/Backends/BackendUtils.cpp

```diff
@@ -81,6 +75,45 @@ runtime::RuntimeBundle::getSymbolInfo(const Named *v) const {
   return it->second;
 }
 
+/// If \p PH is an output placeholder, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool isOutput(const Placeholder *PH) {
+  for (auto &use : PH->getUsers()) {
+    // Look through the inputs of the PH's users. If an input is overwritten
+    // check if it's the PH, if it is return true.
+    for (unsigned i = 0, numInputs = use.getUser()->getNumInputs();
```

Comment:
nit: you use `use.getUser()` multiple times, could just create a local variable for it.